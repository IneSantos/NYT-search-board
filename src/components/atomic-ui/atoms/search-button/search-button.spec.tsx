import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { render } from "@testing-library/react";
import SearchButton from "./search-button";

let store;
let mockedStore;

describe("SearchButton atom tests", () => {
  test(`SearchButton component render`, () => {
    mockedStore = configureStore();
    store = mockedStore({
      article: {
        articleList: [
          {
            id: "",
            headline: "",
            abstract: "",
            keywords: [],
            section: "",
            snippet: "",
            pub_date: "",
            source: "",
            web_url: "",
          },
        ],
        page: 0,
        searchClicked: false,
      },
    });

    const { container } = render(
      <Provider store={store}>
        <SearchButton input="This is an input" />
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
