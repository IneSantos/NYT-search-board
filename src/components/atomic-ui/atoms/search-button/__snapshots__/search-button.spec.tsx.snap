// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton atom tests SearchButton component render 1`] = `
<button
  class="clear-button"
  type="reset"
>
  clear
</button>
`;
